addons/web_editor/static/src/js/editor/odoo-editor/test/spec/insert.test.js

```diff
@@ -94,17 +73,19 @@ describe('insert HTML', () => {
         });
         it('should not unwrap single node if the selection anchorNode is the editable', async () => {
             await testEditor(BasicEditor, {
-                contentBefore: '<p>content</p>[]',
+                contentBefore: '<p>content</p>',
                 stepFunction: async editor => {
+                    setCursorEnd(editor.editable, false);
                     await editor.execCommand('insert', parseHTML('<p>def</p>'));
                 },
                 contentAfter: '<p>content</p><p>def[]</p>',
             });
         });
         it('should not unwrap nodes if the selection anchorNode is the editable', async () => {
             await testEditor(BasicEditor, {
-                contentBefore: '<p>content</p>[]',
+                contentBefore: '<p>content</p>',
                 stepFunction: async editor => {
+                    setCursorEnd(editor.editable, false);
                     await editor.execCommand('insert', parseHTML('<div>abc</div><p>def</p>'));
                 },
                 contentAfter: '<p>content</p><div>abc</div><p>def[]</p>',
```

Comment:
@abd-msyukyu-odoo, I've changed your tests slightly. It seems to reflect your use case based on your comment:

> I use the setCursorEnd method to put the cursor at the end of the editable element, then I insert a fragment with my embedded view block which is a DIV as well as an empty paragraph p br

But I could not find the mentioned piece of code that calls `setCursorEnd` followed by a `insert` command.

Could you please check that:
- this test still reflects the use case you had in mind,
- this PR does not break the mentioned knowledge features?

Thanks!

Reply:
Thanks, the updated test corresponds to what is done in Knowledge:
the code is there: <https://github.com/odoo/enterprise/blob/d1d3f53edea729bf95220eb7ddf303f44194b51f/knowledge/static/src/js/wysiwyg.js#L325-L326>

So for me it's all good!